core/src/main/java/org/apache/calcite/sql/SqlDialectFactoryImpl.java

```diff
@@ -122,7 +122,7 @@ public SqlDialect create(DatabaseMetaData databaseMetaData) {
     case "MYSQL (INFOBRIGHT)":
       return new InfobrightSqlDialect(c);
     case "MYSQL":
-      return new MysqlSqlDialect(c);
+      return new MysqlSqlDialect(c.withDataTypeSystem(MysqlSqlDialect.MYSQL_TYPE_SYSTEM));
```

Comment:
Can we add the same for Postgres? Seems to have same problem.

Nit. Can we push after _(_ to new line?